shanoir-ng-datasets/src/main/java/org/shanoir/ng/importer/service/ImporterService.java

```diff
@@ -167,18 +174,37 @@ public void createAllDatasetAcquisition(ImportJob importJob, Long userId) throws
                 QualityTag tagSave = subjectStudy != null ? subjectStudy.getQualityTag() : null;
                 ExaminationData examData = new ExaminationData(examination);
                 examData.setDatasetAcquisitions(Utils.toList(generatedAcquisitions));
-                QualityCardResult qualityResult = checkQuality(examData, importJob);                				
+                QualityCardResult qualityResult;
+                
+                // If import comes from ShanoirUploader, the check quality at import has already been done
+                if (!importJob.isFromShanoirUploader()) {
+                    qualityResult = qualityService.checkQuality(examData, importJob, null);
+                } else {
+                    LOG.info("Importing Data from ShanoirUploader.");
+                    // We retrieve quality card result from ShUp import job
+                    qualityResult = qualityService.retrieveQualityCardResult(importJob);
+                    if (!qualityResult.isEmpty()) {
+                        LOG.info("Retrieving Quality Control result from ShanoirUploader.");
+                        if(subjectStudy != null) {
+                            subjectStudy.setQualityTag(qualityResult.get(0).getTagSet());
+                            qualityResult.addUpdatedSubjectStudy(subjectStudy);
+                        }
+                    }
+                }
+                                				
                 // Has quality check passed ?
-                if (qualityResult.hasError()) {
+                if (qualityResult != null && !qualityResult.isEmpty() && qualityResult.hasError()) {
+                    // TODO : Delete newly created Examination ?
```

Comment:
I think deleting the exam would not fit to the frontend import workflow as the exam is created "independently"

Reply:
I was worried about the empty created exams if quality control is KO and the fact that the user may recreated a new exam each time he tries to reimport the data.